Update and side bar pipelines overview section (#5969)
circleci/circleci-docs: jekyll/_cci2/skip-build.md

---
layout: classic-docs
title: Skip or Cancel Pipelines
description: This document describes how to prevent automatically building your project.
order: 100
version:
- Cloud
- Server v3.x
- Server v2.x
---

This document describes how to skip or cancel pipelines. There are a couple of ways to do this. A pipeline can be skipped on commit, or by using the auto-cancel feature. Both methods are described below.

* TOC
{:toc}

## Skipping a pipeline
{: #skipping-a-build }

> By default, CircleCI automatically runs a pipeline whenever you push changes to your project. You can override this behavior by adding a `[ci skip]` or `[skip ci]` tag within the first 250 characters of the body of the commit or the commit's title. This not only skips the marked commit, but also **all other commits** in the push. 

Author: ```suggestion
By default, CircleCI automatically runs a pipeline whenever you push changes to your project. You can override this behavior by adding a `[ci skip]` or `[skip ci]` tag within the first 250 characters of the body of the commit or the commit's title. This not only skips the marked commit, but also **all other commits** in the push.
```

**CircleCI server v2.x**
If you are using server v2.x you are not using the pipelines feature but the method of skipping builds described here can still be used.

> ### Scope

Author: ```suggestion
### Scope
{: #scope }
```

---
layout: classic-docs

> title: Skip or Cancel Pipelines

Reviewer: Can we avoid saying "pipelines" here? Maybe something like:

```suggestion
title: Skip and Cancel: avoiding unnecessary credit spend
```

Author: "Skip and cancel jobs and workflows"?

Reviewer: Yeah, referring to jobs and workflows will be correct for now. We will probably eventually move the skip to pipeline level, but we can correct that when it happens.

> description: This document describes how to prevent automatically building your project.

Reviewer: Are we happy to use "building", but not "build"?

Author: This is hard!! something like:
"This document describes the options available to you for controlling when work is automatically carried out on your project, by skipping jobs or auto-cancelling workflows"

Author: I will also add some use case info about preventing unnecessary credit usage.

Reviewer: Sounds good. I think your alternative wording there is fine.

> description: This document describes how to prevent automatically building your project.

Author: ```suggestion
description: This document describes the options available to you for controlling when work is automatically carried out on your project, by skipping jobs or auto-cancelling workflows.
```

order: 100
version:
- Cloud
- Server v3.x
- Server v2.x
---

> This document describes how to skip or cancel pipelines. There are a couple of ways to do this. A pipeline can be skipped on commit, or by using the auto-cancel feature. Both methods are described below.

Reviewer: This needs the "jobs and workflows" treatment

* TOC
{:toc}

> ## Skipping a pipeline

Reviewer: How about?

```suggestion
## Skipping jobs in a pipeline
```

{: #skipping-a-build }

> By default, CircleCI automatically runs a pipeline whenever you push changes to your project. You can override this behavior by adding a `[ci skip]` or `[skip ci]` tag within the first 250 characters of the body of the commit or the commit's title. This not only skips the marked commit, but also **all other commits** in the push.

Reviewer: ```suggestion
By default, CircleCI automatically triggers a pipeline whenever you push changes to your project. You can override this behavior by adding a `[ci skip]` or `[skip ci]` tag within the first 250 characters of the body or title of the commit. This not only skips the marked commit, but also **all other commits** in the push.
```

**CircleCI server v2.x**

> If you are using server v2.x you are not using the pipelines feature but the method of skipping builds described here can still be used.

Reviewer 2: ```suggestion
If you are using server v2.x, you are not using the pipelines feature. The method for skipping builds described here can still be used.
```

### Scope
{: #scope }

A few points to note regarding the scope of the `ci skip` feature:

> * The pipeline and workflows will still exist for these commits but all jobs will **not** be run.

Reviewer 3: That's a bit awkward. How about "no jobs will be run"?

Author: 😆 yes thanks for pointing this out!

* If you push multiple commits at once, a single `[ci skip]` or `[skip ci]` will skip the build **for all commits**.

> * This feature is not supported for fork PRs. Scheduled workflows will not be cancelled even if you push a commit with `[ci skip]` message. Changing the config file is the way to upgrade the current schedule.

Reviewer: Suggest eliminating "double negative": "not be cancelled" == "run"?

```suggestion
* This feature is not supported for fork PRs. Scheduled workflows will run even if you push a commit with `[ci skip]` message. Changing the config file is the way to upgrade the current schedule.
```

When pushed to a VCS, this commit will not be built on CircleCI because of the `[ci skip]` or `[skip ci]` in the commit description.

## Auto cancelling
{: #auto-cancelling}

If you are frequently pushing changes to a branch, you increase the chances of queueing. This means you might have to wait for an older pipeline to complete before the most recent version starts.

To save time, you can configure CircleCI to automatically cancel any queued or running pipelines when a newer pipeline is triggered on that same branch.

> ### Scope

Author: ```suggestion
### Scope
{: #scope }
```

When pushed to a VCS, this commit will not be built on CircleCI because of the `[ci skip]` or `[skip ci]` in the commit description.

## Auto cancelling
{: #auto-cancelling}

If you are frequently pushing changes to a branch, you increase the chances of queueing. This means you might have to wait for an older pipeline to complete before the most recent version starts.

> To save time, you can configure CircleCI to automatically cancel any queued or running pipelines when a newer pipeline is triggered on that same branch.

Reviewer: ```suggestion
To save time, you can configure CircleCI to automatically cancel any non-terminated workflows when a newer pipeline is triggered on that same branch.
```

### Scope
{: #scope }

A few points to note regarding the use of the auto-cancel feature:

> * When using the auto-cancel feature, pipelines will still be created but the included workflows (and therefore jobs) will not run.

Reviewer: ```suggestion
* When using the auto-cancel feature, triggering a pipeline will automatically cancel all workflows (and jobs) from the previous pipeline on the same branch
```

Reviewer 2: ```suggestion
* When using the auto-cancel feature, pipelines will still be created, but the included workflows (and therefore jobs) will not run.
```

> * When using the auto-cancel feature, pipelines will still be created, but the included workflows (and therefore jobs) will not run.

Reviewer: This sentence does not make sense to me. The workflows in the most recent will run, but *previous* workflows are cancelled.

Author: Yeah sorry that does not make sense! I'll just remove the line.
